Added ingestion details and step by step process (#569)
moj-analytical-services/user-guidance: source/documentation/tools/ingestion/index.md

## How the ingestion service works

### Onboarding 

1. You raise an onboarding request by sending us information about the supplier and destination bucket.
1. We check whether this is a standard use case. If the request needs bespoke logging, security rules, transformations, or other non-standard behaviour, it may be outside our current offer.
1. We use the supplier details to create access with an SFTP account and IP addresses.
1. We configure the destination bucket location.
1. We check the bucket's permissions. You may need to update the bucket's policy to allow the ingestion service to copy files into it.  
1. If your destination bucket uses KMS encryption, we use the key you sent us to set up write permissions for the ingestion service.
1. We notify you that the onboarding request has been completed and confirm the username for your account.

### Copying data to an S3 bucket

1. Your supplier connects to the SFTP endpoint with the username we created and SSH key.
1. Your supplier uploads files to their home directory in a landing bucket.
1. Each file triggers a GuardDuty security scan. 

> 1. When GuardDuty finishes scanning, the service moves clean files to your destination bucket and infected files to a quarantine bucket.

Reviewer: Could you add some details on what happens to data that doesn't pass the scan? Where is the bucket? What happens next etc?

Author: Updated, but ultimately there isn't much a user can do - they can't access the quarantine bucket, so I've added in a suggestion to speak to their supplier who uploaded the data.

1. Your team accesses the files in the destination bucket. 

## Onboard to the service

To use the ingestion service, you'll need to give us the following:

- supplier's name (this can be a person or organisation)
- supplier's email
- supplier's IP address(es)
- supplier's SSH public key
- destination bucket's location on Analytical Platform (for example: `s3://${DESTINATION_BUCKET}/${OPTIONAL_PREFIX}`)
- destination bucket's KMS key if encrypted

Please [raise a support ticket](https://github.com/ministryofjustice/data-platform-support/issues/new?template=analytical-platform-ingestion.yml) with the required information to start the onboarding process.

> > **Note**: We'll use the supplier's name as your `USERNAME`.

Reviewer 2: I'm not sure this is completely correct. See this [thread](https://moj.enterprise.slack.com/archives/C4PF7QAJZ/p1776885719629179?thread_ts=1775038578.288189&cid=C4PF7QAJZ) - sometimes we will need to alias

Author: Updated to reflect this!

1. Your team accesses the files in the destination bucket. If any files are in the quarantine bucket, speak to your supplier.

## Onboard to the service

To use the ingestion service, you'll need to give us information about the supplier using the service and the destination bucket. We need contact information of an individual to send notifications and updates about the ingestion service. 

[Raise a support ticket](https://github.com/ministryofjustice/data-platform-support/issues/new?template=analytical-platform-ingestion.yml) with the following information to start the onboarding process:

- supplier's name (this should be a specific person)
- supplier's email (not a generic inbox)
- supplier's IP addresses
- supplier's SSH public key
- destination bucket's location on Analytical Platform (for example: `s3://${DESTINATION_BUCKET}/${OPTIONAL_PREFIX}`)
- destination bucket's KMS key, if encrypted
- an email address or Slack channel to send scan completion alerts to (optional)

> > **Note**: We'll generally use the supplier's name or organisation as the account's `USERNAME`, which the person or system using the ingestion service needs to log in. We may change it if we think there's a security issue.

Author: ```suggestion
> **Note**: We'll generally use the supplier or organisation name as the account's `USERNAME`, which the person or system using the ingestion service needs to log in. As this is stored publicly in code, we may ask you for an alternative name to avoid exposing sensitive information.
```

1. We use the supplier details to create access with an SFTP account and IP addresses.
1. We configure the destination bucket location.
1. We check the bucket's permissions. You may need to update the bucket's policy to allow the ingestion service to copy files into it.  
1. If your destination bucket uses KMS encryption, we use the key you sent us to set up write permissions for the ingestion service.
1. We notify you that the onboarding request has been completed and confirm the username for your account. 

### Copying data to an S3 bucket

1. Your supplier connects to the SFTP endpoint with the username we created and SSH key.
1. Your supplier uploads files to their home directory in a landing bucket.
1. Each file triggers a GuardDuty security scan. 
1. When GuardDuty finishes scanning, you'll receive a notification by Slack or email if you requested this The service moves clean files to your destination bucket and infected files to a quarantine bucket. You cannot access files in the quarantine bucket. 
1. Your team accesses the files in the destination bucket. If any files are in the quarantine bucket, speak to your supplier.

## Onboard to the service

> To use the ingestion service, you'll need to give us information about the supplier using the service and the destination bucket. We need contact information of an individual to send notification and updates about the ingestion service. 

Author: ```suggestion
To use the ingestion service, you'll need to give us information about the supplier using the service and the destination bucket. We need contact information of an individual to send notifications and updates about the ingestion service. 
```

[Raise a support ticket](https://github.com/ministryofjustice/data-platform-support/issues/new?template=analytical-platform-ingestion.yml) with the following information to start the onboarding process:

- supplier's name (this should be a specific person)
- supplier's email (not a generic inbox)
- supplier's IP addresses
- supplier's SSH public key
- destination bucket's location on Analytical Platform (for example: `s3://${DESTINATION_BUCKET}/${OPTIONAL_PREFIX}`)
- destination bucket's KMS key, if encrypted
- an email address or Slack channel to send scan completion alerts to (optional)

Please [raise a support ticket](https://github.com/ministryofjustice/data-platform-support/issues/new?template=analytical-platform-ingestion.yml) with the required information to start the onboarding process.

> 

Author: ```suggestion

```
